File: PayJunction Webhook/PayJunction/PayJunction/Transaction/GetTransactionRequest.vb
Imports RestSharp
Imports Newtonsoft.Json

Namespace PayJunction.Transaction
    Class GetTransactionRequest
        Inherits Request(Of TransactionResponse)
        Public Overrides ReadOnly Property EndPoint() As String
            Get
                Return Convert.ToString("transactions/") & TransactionID
            End Get
        End Property

        Public Overrides ReadOnly Property Method() As Method
            Get
                Return Method.[GET]
            End Get
        End Property

        Public Overrides Function ProcessResponse() As TransactionResponse
            Dim resp As TransactionResponse = JsonConvert.DeserializeObject(Of TransactionResponse)(RawResponse)
            Return resp
        End Function

        Public Property TransactionID() As String
            Get
                Return m_TransactionID
            End Get
            Set
                m_TransactionID = Value
            End Set
        End Property
        Private m_TransactionID As String

        Public Sub New(id As String)
            Me.TransactionID = id
        End Sub
    End Class
End Namespace

File: PayJunction C#/ConsoleApplication2/PayJunction/Transaction/GetTransactionRequest.cs
using PayJunction;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RestSharp;
using System.Net.Http;
using Newtonsoft.Json;

namespace PayJunction.Transaction
{
    class GetTransactionRequest : Request<TransactionResponse>
    {
        public override string EndPoint
        {
            get
            {
                return "transactions/"+TransactionID;
            }
        }

        public override Method Method
        {
            get
            {
                return Method.GET;
            }
        }

        public override TransactionResponse ProcessResponse()
        {
            TransactionResponse resp = JsonConvert.DeserializeObject<TransactionResponse>(RawResponse);
            return resp;
        }

        public string TransactionID
        {
            get;set;
        }

        public GetTransactionRequest(string id)
        {
            this.TransactionID = id;
        }
    }
}
